### DISCUSSION TITLE
Improve Release Process / Weblate integration

### DISCUSSION THREAD

**Discussion by pljones** [jamulussoftware/jamulus#3140] [Ideas]
**What is the current behaviour and why should it be changed?**

The current release process documentation makes little reference to how either app or website documentation is managed through Weblate, or how a someone handling the release process can check who has been making changes to a translation on Weblate (or, indeed, directly via git).

**Describe possible approaches**

Ideally, the tooling would be able to maintain the current Weblate translators without intervention and we would _require_ that a translator had a visible Github identity associated with the Weblate (or git) contributions.

In any case, improved documentation on how translations work is needed or needs to be referened with clear guidance in the release process documentation.

**Has this feature been discussed and generally agreed?**

Raised concerns over in https://github.com/jamulussoftware/jamulus/issues/3126.  Closing that issue based on raising this.

**Comment by ann0see** [comment]
@ignotus666 knows more about the Weblate process. I think the mess was created by the website having conflicts.

**Comment by ann0see** [comment]
> we would require that a translator had a visible Github identity associated with the Weblate (or git) contributions.

Unfortunately this isn't possible. Everyone with a weblate account can make changes to that. Therefore the Weblate PRs should be checked by the translators.

**Comment by pljones** [comment]
> Therefore the Weblate PRs should be checked by the translators.

...who would be required to have visible Github identities, so they could sign off their check by closing the raised issue.

**Comment by ann0see** [comment]
Kind of. Yes. But there are no guarantees that that's the case.

**Comment by pljones** [comment]
What I'm saying is that there's no point us raising the current issues _unless that is guaranteed to be the case_.  So either we change policy to require it or we drop creating the issues at all and just rely on Weblate.  There's no point trying to maintain something inherently broken.

**Comment by ann0see** [comment]
The issues are for notifying translators to start translating. Maybe we can condense it to one issue which pings all translators. That reduces complexity

**Comment by pljones** [comment]
If the translation is being managed through Weblate, we should have a way on Weblate of notifying the translators.  The Github management shouldn't be related.

**Comment by ann0see** [comment]
Weblate does have a notification system and I did use it to notify the translators. 

However, there might be translators who are on GitHub and Weblate. 

Discussing the translation process is inconvenient on Weblate in my opinion. 

I think @jamulussoftware/translators should judge how we handle translation in future.

**Reply by pljones** [reply]
Converted to a discussion until we have something that can be actioned.

**Comment by ann0see** [comment]
I think we should revisit this after the release got out.

**Comment by ann0see** [comment]
Ok. One thing: https://hosted.weblate.org/projects/jamulus/#reports shows the translators in a given timeframe. I think this should be added to the documentation.

**Reply by pljones** [reply]
It's also possible to add a per-language link to the app tickets.  (Trickier for the website, as it's multiple Weblate components - there may be a query syntax for "everything except the app", I didn't check.)

e.g.
> Progress: https://hosted.weblate.org/changes/browse/jamulus/jamulus-app/sv_SE/?start_date=2023-07-30

**Comment by ann0see** [comment]
Ok. @ignotus666 @comradekingu (as you know Weblate)

@pljones Can we make a list for what needs to be done here?
I believe:
1. The release checklist must be updated - e.g what do we need to do on Weblate? 
2. The translator tooling (create translation issues, change log script,...) needs to be updated: Use the Weblate API
3. Documentation for the website needs to be updated and added

**Reply by pljones** [reply]
I don't know what's needed on the Weblate side.  What's need in the Release Process is an explanation of what guidance may need to be given on the Weblate side to new translators, for example - where to find things, what to do, what not to do.  Perhaps that needs a new document.

Again, I don't know what Weblate offers as an API or what degree of integration is possible.  The change I added to the app template was simple because it's just an embedded URL.

And I'm also not that familiar with the website documentation - I've done only a little there.  I'm not sure much is really needed - but I didn't feel confident around the website processes.

Basically, not knowing what's needed means I can't really say what would help.  Just "better" documentation, written from the point of view of enabling someone who knows _nothing_ about putting out a release to do the job effectively.  Essentially, this could apply to the whole Release Process document, not just Weblate integration, of course.

**Reply by ann0see** [reply]
> What's need in the Release Process is an explanation of what guidance may need to be given on the Weblate side to new translators, for example - where to find things, what to do, what not to do.

Basically an updated TRANSLATING.md file

**Reply by ann0see** [reply]
> Just "better" documentation, written from the point of view of enabling someone who knows nothing about putting out a release to do the job effectively. Essentially, this could apply to the whole Release Process document, not just Weblate integration, of course.

That's what the checklist should do...

**Reply by pljones** [reply]
> Basically an updated TRANSLATING.md file

Isn't that aimed at translators?  The Release Process is aimed at the person managing the release, so we shouldn't try to muddy the focus of TRANSLATING.md unnecessarily.

> That's what the checklist should do...

OK, but someone who doesn't know what _should_ be done can't suggest improvements, only call out places where it fell short of that goal.

**Reply by ann0see** [reply]
TRANSLATING.md: I thought that we needed to update the Translation for Translators too, not sure if it's good enough at the moment? Can some translator comment on that?
Checklist: Agree. Do you have a specific issue we can fix?

**Reply by pljones** [reply]
* For each supported platform, there should be a clearly documented process of how to build and package the installer.  This isn't currently the case.
  **Reason:** Should we need to switch from Github at a time when the existing workflows are no longer understood by anyone on the team, having a clear explanation of what needs doing will ease migration.
  * This need not be in Release-Process.md but there should be links there so the documentation is easily found.
  * In addition, the differences in building final, release candidate, beta and development builds should be documented somewhere, too, along with how those affect the tooling.

* We need to be clear about when `lrelease` should be run -- should new translation files be included with a PR, or should we postpone all translation file generation until the release?
  I'd favour iterative development, myself -- but then, I'm not a translator: I just want Jamulus to be ready for release on PR merge 😄.

* The issue template for the release and Release-Process.md need reviewing to make sure they're in agreement.
  * For example, "Notify all the translators that translation is required" in Release-Process.md should also mention updating the script with current translators (if we're going to continue using it).

* The checklist for translations isn't clear: for example, "Punctuation and spacing consistent" - this should reference some guideline documentation.

* We may want to review the SourceForge processes in the light of recent discoveries.

* The whole of the website section needs reviewing by someone who knows the processes.

* The announcement template could be a Github issue template
  * I also don't like the layout - the one we used for this release was nicer, so it could do with Release-Process.md being updated.

* Speaking of issue template layouts, I like the planned dates in the "Prepare Release \<rel\>" issue tabular, too ([like the current one](https://github.com/jamulussoftware/jamulus/issues/3094)).

* The ChangeLog generator seems to "generally work" - if there are common points where "manual assistance is require", we should aim to capture them -- I think that might be worth adding to the Release-Process.md section about using the script.

**Reply by ann0see** [reply]
Thank you! This sounds like a good list.

**Reply by ann0see** [reply]
> We may want to review the SourceForge processes in the light of recent discoveries.

I believe we can only test that if we do a release. 

> The whole of the website section needs reviewing by someone who knows the processes.

Which leaves me, @ignotus666 and @gilgongo 

> The announcement template could be a Github issue template

We had this discussion already, but it might confuse people who open an issue. 

> The ChangeLog generator seems to "generally work" - if there are common points where "manual assistance is require", we should aim to capture them -- I think that might be worth adding to the Release-Process.md section about using the script.

Agree. Not sure if we can count on hoffie being available.

**Reply by pljones** [reply]
> We had this discussion already, but it might confuse people who open an issue.

Mmm...  I don't think the template's current location is ideal, though.

**Comment by jujudusud** [comment]
I had this [discussion comment](https://github.com/orgs/jamulussoftware/discussions/3171#discussioncomment-6983667) in the release retrospective.
What do you think about this point of view ?
